### PR TITLE
[tx] move logits computation LogitsProcessorMixin and fix prompt_logprobs

### PR DESCRIPTION
**Pull request by raulchen**
## Motivation
  Different scenarios have different requirements for logits computation:

  - **Training**: Only need logprobs of target tokens, not full `[B, T, V]` logits
  - **Prefill**: Only need last position logits for sampling, plus optional prompt logprobs
  - **Decode**: Need full `[B, 1, V]` logits for sampling transforms (temperature, top_k, top_p)

  By returning `hidden_states` and letting callers decide what to compute, we avoid unnecessary materialization of
  full vocabulary logits. This also enables future chunked logprobs computation for memory efficiency.
 
 ## Changes

  - Add `LogitsProcessorMixin` in `tx/utils/logits_processor.py` with:
    - Abstract `get_lm_head()` method
    - `compute_logits()`, `compute_logprobs()`, `logits_to_logprobs()`
  - Remove `logits` from `CausalLMOutput` - callers now explicitly call `compute_logits()`
  - Remove `skip_prompt_logits` parameter from model forward
  - Add validation: `train_unembed=True` incompatible with `tie_word_embeddings=True`

  ## Code consolidation

  Logits/logprobs computation was scattered across multiple locations:
  - `tx/layers/logits_processor.py` (deleted)
  - `tx/utils/generator.py` (`compute_prompt_logprobs()` helper, manual log_softmax)
  - `tx/tinker/backends/jax.py` (manual logprobs from logits)

  Now unified in `LogitsProcessorMixin` with a single implementation.

  ## Bug fix

  The prompt_logprobs computation was not passing adapter_indices to compute_logprobs, causing incorrect results
  with LoRA adapters. Added test coverage.

### CODE REVIEW

File: skyrl-tx/tx/models/llama3.py [hunk 1]
```diff
@@ -262,16 +262,15 @@ def __call__(
...
+class Llama3ForCausalLM(nnx.Module, GeneratorMixin, CausalLMBase):
 
     def __init__(self, config: LlamaConfig, *, dtype: jnp.dtype, rngs: nnx.Rngs) -> None:
-        self.config = config
         self.model = Llama3Model(config, dtype=dtype, rngs=rngs)
 
-        if self.config.tie_word_embeddings:
-            self.lm_head = self.model.embed_tokens.T
+        if config.tie_word_embeddings:
+            lm_head = self.model.embed_tokens.T
```
[the remark this answers is not in the document]
Comment (raulchen) [reply skyrl-tx/tx/models/llama3.py]: this is an existing bug / TODO. 
added an assertion to prevent enabling tie_word_embedding and train_unembed together.

File: skyrl-tx/tx/tinker/backends/jax.py [hunk 2]
```diff
@@ -229,22 +229,28 @@ def _jit_timing_context(self, seq_len: int, mode: str):
     def _create_loss_and_grad_fn(self):
         """Compile and cache the loss function to avoid re-jitting on every call."""
 
-        def _model_forward(
+        def _forward_and_logprobs(
```
Comment (pcmoritz) [inline skyrl-tx/tx/tinker/backends/jax.py]: Just one small nit: Can we keep the original `_model_forward` name? I think actually the way it is done now (including computing the softmax) is more natural for a `forward`, so there is no need to point out the fact that we compute the logprobs.

**Review by pcmoritz — APPROVED** [review]
